homeassistant/components/imeon_inverter/select.py

```diff
@@ -0,0 +1,70 @@
+"""Imeon inverter select support."""
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+import logging
+
+from homeassistant.components.select import SelectEntity, SelectEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from .const import ATTR_INVERTER_MODE, INVERTER_MODE_OPTIONS
+from .coordinator import Inverter, InverterCoordinator
+from .entity import InverterEntity
+
+type InverterConfigEntry = ConfigEntry[InverterCoordinator]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class ImeonSelectEntityDescription(SelectEntityDescription):
+    """Class to describe an Imeon inverter select entity."""
+
+    set_value_fn: Callable[[Inverter, str], Awaitable[bool]]
+
+
+SELECT_DESCRIPTIONS: tuple[ImeonSelectEntityDescription, ...] = (
+    ImeonSelectEntityDescription(
+        key="manager_inverter_mode",
+        translation_key="manager_inverter_mode",
+        options=list(INVERTER_MODE_OPTIONS),
+        set_value_fn=lambda api, mode: api.set_inverter_mode(
+            INVERTER_MODE_OPTIONS[mode]
+        ),
```

Comment:
It's still not clear to me what the library expects. If it's `smg`, you should map `smg` to `smart_grid` or something, and then use translations to make that `Smart Grid`